packages/formik/src/useFormik.tsx

```diff
@@ -0,0 +1,512 @@
+  });
+
+  const validateForm = useEventCallback(() => {
+    throw new Error('not implemented');
+  });
+
+  const validateField = useEventCallback(() => {
+    throw new Error('not implemented');
+  });
+
+  const setStatus = useEventCallback(() => {
+    throw new Error('not implemented');
+  });
+
+  const submitForm = useEventCallback(() => {
+    throw new Error('not implemented');
+  });
+
+  const resetForm = useEventCallback(() => {
+    throw new Error('not implemented');
+  });
+
+  // @todo we may need ResetButton or useResetButton()
+  // const handleReset = useEventCallback(() => {
+  //   throw new Error('not implemented');
+  // });
+
+  const setFormikState = useEventCallback(() => {
+    throw new Error('not implemented');
+  });
+
+  // @todo
+  // const handleFocus = useEventCallback(() => {
+  //   throw new Error('not implemented');
+  // });
+  // const setFocus = useEventCallback(() => {
+  //   throw new Error('not implemented');
+  // });
+
+  // @todo fake these?
+  // const getFieldProps = useEventCallback(() => {
+  //   throw new Error('not implemented');
+  // });
+  // const getFieldMeta = useEventCallback(() => {
+  //   throw new Error('not implemented');
+  // });
+  // const getFieldHelpers = useEventCallback(() => {
+  //   throw new Error('not implemented');
+  // });
+
+  const handleSubmit = useEventCallback(async e => {
+    if (e.preventDefault) {
+      e.preventDefault();
+    }
+
+    // set submitting to true
+    setSubmitting(true);
+
+    let fieldErrors: any = {};
+    let formValidateErrors;
+    let formSchemaErrors;
+
+    // validate fields
+    forEachField(fieldRegistry, (key, field) => {
+      field.setTouched(true);
+      if (field.validate) {
+        fieldErrors[key] = field.validate(field.valueRef.current);
+      }
+    });
+
+    // @todo extract and implement validateForm and validateField
+    if (validate) {
+      formValidateErrors = validate(getValues());
+    }
+
+    if (validationSchema) {
+      try {
+        validateYupSchema(getValues(), validationSchema, true);
+      } catch (err) {
+        // Yup will throw a validation error if validation fails. We catch those and
+        // resolve them into Formik errors. We can sniff if something is a Yup error
+        // by checking error.name.
+        // @see https://github.com/jquense/yup#validationerrorerrors-string--arraystring-value-any-path-string
+        if (err.name === 'ValidationError') {
+          formSchemaErrors = yupToFormErrors(err);
+        } else {
+          // We throw any other errors
+          if (process.env.NODE_ENV !== 'production') {
+            console.warn(
+              `Warning: An unhandled error was caught during validation in <Formik validationSchema />`,
+              err
+            );
+          }
+        }
+      }
+    }
+
+    const combinedErrors = deepmerge.all<FormikErrors<Values>>(
+      [fieldErrors, formSchemaErrors, formValidateErrors],
```

Comment:
This will throw an error in `deepmerge.all` trying to convert `undefined` to an object as `formSchemaErrors` and `formValidateErrors` don't always get set.